src/components/document/document-content.tsx

```diff
@@ -53,25 +54,41 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   private domElement: HTMLElement | null;
   private rowRefs: Array<TileRowComponent | null>;
   private mutationObserver: MutationObserver;
+  private scrollDisposer: IReactionDisposer;
 
   public componentDidMount() {
     if (this.domElement) {
       this.domElement.addEventListener("scroll", throttle(this.updateVisibleRows, 100));
       this.updateVisibleRows();
       this.scrollToSection(this.props.selectedSectionId);
 
-      if ((window as any).MutationObserver) {
-        this.mutationObserver = new MutationObserver(this.handleRowElementsChanged);
-        this.mutationObserver.observe(this.domElement, { childList: true });
-      }
       // We pass the domElement to our children, but it's undefined during the first render,
       // so we force an update to make sure we draw at least once after we have our domElement.
       this.forceUpdate();
+
+      this.scrollDisposer = reaction(() => this.stores.ui.scrollToTileId,
+        (tileId: string | undefined, prevTileId: string | undefined) => {
+        if (tileId) {
+          this.rowRefs.forEach((row: TileRowComponent | null) => {
+            if (row?.tileRowDiv && row.hasTile(tileId)) {
```

Comment:
If I remember right the tile ids of curriculum "docs" are not unique. So when scrolling to a tile in a curriculum doc, other docs (not visible) docs might try to scroll too. Perhaps that will be fine because they aren't visible, but it might be worth checking.